handlers/base.go

```diff
@@ -238,3 +241,48 @@ func SplitMsg(text string, max int) []string {
 
 	return parts
 }
+
+// NewTelQueryReceiveHandler creates a new receive handler given the passed in text and from fields
+func NewTelQueryReceiveHandler(h BaseHandler, fromField string, textField string) courier.ChannelHandleFunc {
+	// ReceiveMessage is our HTTP handler function for incoming messages
+	return func(ctx context.Context, c courier.Channel, w http.ResponseWriter, r *http.Request) ([]courier.Event, error) {
+		text := r.URL.Query().Get(textField)
+		from := r.URL.Query().Get(fromField)
+		if from == "" {
+			return nil, courier.WriteAndLogRequestError(ctx, w, r, c, fmt.Errorf("missing required field '%s'", fromField))
+		}
+		// create our URN
+		urn := urns.NewTelURNForCountry(from, c.Country())
+
+		// build our msg
+		msg := h.Backend().NewIncomingMsg(c, urn, text).WithReceivedOn(time.Now().UTC())
+
+		// and finally queue our message
+		err := h.Backend().WriteMsg(ctx, msg)
+		if err != nil {
+			return nil, err
+		}
+
+		return []courier.Event{msg}, courier.WriteMsgSuccess(ctx, w, r, []courier.Msg{msg})
+	}
+}
+
+// GetTextAndAttachments returns both the text of our message as well as any attachments, newline delimited
+func GetTextAndAttachments(m courier.Msg) string {
```

Comment:
Moved this and the one below to handlers since thats the only place they are used.